src/gti.jl

```diff
@@ -0,0 +1,98 @@
+function time_intervals_from_gtis(gtis::AbstractMatrix{<:Real}, segment_size::Real;
+                                  fraction_step::Real=1, epsilon::Real=1e-5)  
+    spectrum_start_times = Float64[]
+
+    gti_low = @view gtis[:,1]
+    gti_up = @view gtis[:,2]
+
+    for (g1,g2) in zip(gti_low,gti_up)
+        if g2 - g1 + epsilon < segment_size
+            continue
+        end
+
+        newtimes = range(g1, g2 - segment_size + epsilon, step=Float64(segment_size) * fraction_step)
+        append!(spectrum_start_times,newtimes)
+    end
+    return spectrum_start_times, spectrum_start_times .+ segment_size
+end
+
+function calculate_segment_bin_start(startbin::Integer, stopbin::Integer,
+                                     nbin::Integer; fraction_step::Real=1)
+    st = floor.(range(startbin, stopbin, step=Int64(nbin * fraction_step)))
+    if st[end] == stopbin
+        st = st[1:end-1]
+    end
+    if st[end] + nbin > stopbin
+        return st[1:end-1]
```

Comment:
I think this is equivalent to
```suggestion
        pop!(st)
    end
    if st[end] + nbin > stopbin
        pop!(st)
```
right?

Also,  I [have already mentioned](https://github.com/matteobachetti/Stingray.jl/pull/2#discussion_r905500295) that slicing creates copies, so in `st = st[1:end-1]` you are creating a copy of `st[1:end-1]` before reassigning it to the `st` identifier.  [`pop!`](https://docs.julialang.org/en/v1/base/collections/#Base.pop!) simply removes the last element of the collection, without assuming anything about indexing of the array (Julia is 1-based by default, but arbitrary indices are allowed).